apps/angular/src/core/utils/validators.ts

```
@@ -0,0 +1,27 @@
+import { AbstractControl, FormGroup, ValidationErrors, ValidatorFn } from '@angular/forms';
+
+export namespace AppValidators{
+
+	export const MIN_PASSWORD_LENGTH = 8;
+
+	/** Check if the entered passwords match. */
+	export function passwordRepetition(): ValidatorFn {
+		return (group: AbstractControl): ValidationErrors | null => {
+			if (group instanceof FormGroup) {
+				const { password, repeatedPassword } = group.controls;
+
+				if (repeatedPassword.value.length < MIN_PASSWORD_LENGTH) {
```

I don't think this validator should check the length of the password. This is not it's responsibility.

---

Then it is better to check the error about the length in the errors? I need to throw a mismatch error when the password has at least the minimum length

---

I decided to add a mismatch error only when the required password length is reached, because when the repeated password is not long enough, it will obviously not match the first password. Also, an error about insufficient length is a higher priority.

---

From the point of business logic it's not correct. There is a straightforward task: we need to check if passwords are the same. If you don't want to show "Password are different" error while the repeated password length is less than `MIN_PASSWORD_LENGTH` you should handle it on UI directly. 
Also, as you said `minLength` error has higher priority so it should as you want it without any changes. If `repeatedPassword.length < MIN_PASSWORD_LENGTH`, the control has 2 errors: insufficientLength and differentPassword. But since min length has higher priority we'll show only this error.